src/components/StartPrjModal/CreateUsingProjectTemplate/index.tsx

```diff
@@ -0,0 +1,229 @@
+import Modal from 'components/Modal';
+import { FC, MouseEvent, useEffect, useState } from 'react';
+import getRandomString from 'utils/getRandomString';
+import clsx from 'clsx';
+import { useNavigate } from 'react-router-dom';
+import styles from './index.module.scss';
+import gif from './deleteme.gif';
+import CreatePrjModal from '../CreatePrjModal';
+import {
+  fetchAllProjectTemplates,
+  useCreateProjectFromTemplate,
+  useCreateTasksFromProjectTemplate,
+} from './hooks';
+
+interface ICreateUsingProjectTemplateProps {
+  onClose: () => void;
+  onNext: () => void;
+  orgId: number;
+}
+
+const CreateUsingProjectTemplate: FC<ICreateUsingProjectTemplateProps> = ({
+  onClose,
+  onNext,
+  orgId,
+}) => {
+  const navigate = useNavigate();
+  const [showProjectCreate, setShowProjectCreate] = useState(false);
+  const [currentTemplate, setCurrentTemplate] = useState<any>(null);
+  const [projectData, setProjectData] = useState<any>(null);
+
+  useEffect(() => {
+    if (projectData) {
+      createProjectTasksFromTemplate();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [projectData]);
+
+  const useFetchProjectTemplates = fetchAllProjectTemplates(setCurrentTemplate);
+  const createProjectFromTemplate =
+    useCreateProjectFromTemplate(setProjectData);
+  const createTasksFromProjectTemplate = useCreateTasksFromProjectTemplate();
+
+  const filterSelectedTemplate = (selectedTemplate: any) =>
+    setCurrentTemplate(selectedTemplate);
+
+  if (showProjectCreate) {
+    return <CreatePrjModal onClose={onClose} onNext={onNext} orgId={orgId} />;
+  }
+
+  const createTasksFromProjectTemplateFunc = () => {
+    if (!projectData) return [];
```

Comment:
If project is not created from template data then we return empty array. Can you tell more about this thing? Why empty array 🤔